src/transport/UnauthenticatedSessionTable.h

```diff
@@ -118,20 +141,50 @@ class UnauthenticatedSessionTable
     ~UnauthenticatedSessionTable() { mEntries.ReleaseAll(); }
 
     /**
-     * Get a session given the peer address. If the session doesn't exist in the cache, allocate a new entry for it.
+     * Get a responder session with the given ephemeralInitiatorNodeID. If the session doesn't exist in the cache, allocate a new
+     * entry for it.
      *
-     * @return the session found or allocated, nullptr if not found and allocation failed.
+     * @return the session found or allocated, or Optional::Missing if not found and allocation failed.
      */
     CHECK_RETURN_VALUE
-    Optional<SessionHandle> FindOrAllocateEntry(const PeerAddress & address, const ReliableMessageProtocolConfig & config)
+    Optional<SessionHandle> FindOrAllocateResponder(NodeId ephemeralInitiatorNodeID, const ReliableMessageProtocolConfig & config)
     {
-        UnauthenticatedSession * result = FindEntry(address);
+        UnauthenticatedSession * result = FindEntry(UnauthenticatedSession::SessionRole::kResponder, ephemeralInitiatorNodeID);
         if (result != nullptr)
             return MakeOptional<SessionHandle>(*result);
 
-        CHIP_ERROR err = AllocEntry(address, config, result);
+        CHIP_ERROR err = AllocEntry(UnauthenticatedSession::SessionRole::kResponder, ephemeralInitiatorNodeID, config, result);
+        if (err == CHIP_NO_ERROR)
+        {
+            return MakeOptional<SessionHandle>(*result);
+        }
+        else
+        {
+            return Optional<SessionHandle>::Missing();
+        }
+    }
+
+    CHECK_RETURN_VALUE Optional<SessionHandle> FindInitiator(NodeId ephemeralInitiatorNodeID)
+    {
+        UnauthenticatedSession * result = FindEntry(UnauthenticatedSession::SessionRole::kInitiator, ephemeralInitiatorNodeID);
+        if (result != nullptr)
+        {
+            return MakeOptional<SessionHandle>(*result);
+        }
+        else
+        {
+            return Optional<SessionHandle>::Missing();
+        }
+    }
```

Comment:
We need expiry logic, so that we don't have an attacker sending 20 "first messages" that never have any session handling occur, and it takes all session establishment slots.

Reply:
The LRU is already implemented. In `UnauthenticatedSessionTable::AllocEntry`, we use `FindLeastRecentUsedEntry` to locate and reuse an entry when the pool is exhaused.